login-service/src/api-login/dto/create-update-auth-client.dto.ts

```diff
@@ -0,0 +1,72 @@
+import { HttpException, HttpStatus } from "@nestjs/common";
+import { ApiProperty } from "@nestjs/swagger";
+
+/**
+ * Input to `POST /login/client` and PUT /login/client/:id`
+ * Contains data to create or modify a auth client
+ */
+export class CreateOrUpdateAuthClientInput {
+    /**
+     * A list of url strings containing at least one url.
+     * These are the URLs the oauth autorize endpoint will redirect back to
+     *
+     * Defaults to `[]` on create
+     * @example ["https://example.com/oauth/callback?query=value"]
+     */
+    redirectUrls?: string[];
+
+    /**
+     * If given, sets the validity flag of the auth client.
+     *
+     * Defaults to `true` on create
+     * @example true
+     */
+    isValid?: boolean;
+
+    /**
+     * If given, setns the need for the client to authenticate using a secret
+     *
+     * Defaults to `true` on create
+     * @example false
+     */
+    requiresSecret?: boolean;
+
+    /**
+     * Checks a given `CreateOrUpdateAuthClientInput` for validity.
+     *
+     * Valid, if:
+     * - `redirectUrls` is not given or an array of at least one url
+     * - `isValid` is not given or a boolean
+     * - `requiresSecret` is not given or a boolean
+     * @param input The input instance to check
+     * @returns The given instance unchanged
+     */
+    static check(input: CreateOrUpdateAuthClientInput): CreateOrUpdateAuthClientInput {
+        if (
+            input.redirectUrls != undefined &&
+            (!(input.redirectUrls instanceof Array) || input.redirectUrls.length <= 0)
```

Comment:
use `== 0` for unsigned comparison

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/length#description
it is a [precise integers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) with the number of utf16 units. 0 is explicitly given and no string can be shorter than 0 utf16 units